### PR TITLE
Move platform build steps to beats packer

### PR DESCRIPTION
**Pull request by ruflin**
For each platform that beats packer builds a binary, it has additional knowledge about where to store config files and data. For this each platform has its own configuration file. The generic copy / paste part of these config file adjustments was now moved to packer. There is still a `build-before` step call which is made to each beat to make adjustments to the config file which are beat specific. For this each beat still needs platform about the different platforms. Hopefully in the future with data and config directory, this special code can be removed.

### CODE REVIEW

File: dev-tools/packer/xgo-scripts/before_tsg_build.sh [hunk 1]
```diff
@@ -1,12 +0,0 @@
-#!/bin/sh
```
Comment (ruflin) [inline dev-tools/packer/xgo-scripts/before_tsg_build.sh]: @tsg Was this file still needed?

Comment (tsg) [inline dev-tools/packer/xgo-scripts/before_tsg_build.sh]: No, I think it's not used anymore.

File: dev-tools/packer/xgo-scripts/packetbeat_before_build.sh [hunk 2]
```diff
@@ -1,10 +0,0 @@
-#!/bin/sh
-
-PB_PATH=/go/src/github.com/elastic/beats/
-
-if [ -d $PB_PATH ]; then
-	cd $PB_PATH
-	patch -p1 < /gopacket_pcap.patch
```
Comment (tsg) [inline dev-tools/packer/xgo-scripts/packetbeat_before_build.sh]: This (the patch) shouldn't be lost, though. Did you you move it anywhere?

Comment (tsg) [inline dev-tools/packer/xgo-scripts/packetbeat_before_build.sh]: Nevermind, I saw it. That kind of works.

File: dev-tools/packer/xgo-scripts/before_build.sh [hunk 3]
```diff
@@ -1,9 +1,35 @@
...
+# binary
+cp $BEATNAME.yml $PREFIX/$BEATNAME-binary.yml
+# darwin
+cp $BEATNAME.yml $PREFIX/$BEATNAME-darwin.yml
+# win
+cp $BEATNAME.yml $PREFIX/$BEATNAME-win.yml
+
+
+# Contains beat specific adjustments. As it is platform specific knowledge, it should be in packer not the beats itself
+PREFIX=$PREFIX make adjust-cfg
```
Comment (ruflin) [inline dev-tools/packer/xgo-scripts/before_build.sh]: The idea here, is that the knowledge inside the beat about different platforms is as minimal as possible so a new platform to compile could be added without affecting the beats. So far I didn't find a solution that on the one hand works for all beats but still allows beat specific changes. Moving all the beats specifics to packer is not an option because of the community beats. So the current state is that such adjustments are made in the adjust-cfg function. Probably a better name could be `before-build`?

@tsg @andrewkroh Any thoughts on the above?

Comment (ruflin) [inline dev-tools/packer/xgo-scripts/before_build.sh]: Additional note: I would hope that the platform specific things get obsolete when we have a config and data directory.

Comment (tsg) [inline dev-tools/packer/xgo-scripts/before_build.sh]: +1 for `before-build` that gives us more freedom.

**Comment by tsg** [comment]
+1 to this, it's a step forward IMHO.

**Comment by ruflin** [comment]
@tsg I rebased and update the PR accordingly.
